server/src/test/java/org/opensearch/snapshots/RestoreServiceIntegTests.java

```diff
@@ -0,0 +1,241 @@
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.common.CheckedConsumer;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.fs.FsRepository;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class RestoreServiceIntegTests extends OpenSearchSingleNodeTestCase {
+    // TODO there is certainly a better way to do this, but I don't know what it is....
+    public void testRestoreToNewWithAliasAndRename() throws Exception {
+        __testRestoreWithRename(false, false, true, true, true);
+    }
+
+    public void testRestoreToNewWithoutAliasAndWithRename() throws Exception {
+        __testRestoreWithRename(false, false, false, true, true);
+    }
+
+    public void testRestoreToNewWithAliasAndWithoutRename() throws Exception {
+        __testRestoreWithRename(false, false, true, false, false);
+    }
+
+    public void testRestoreToNewWithoutAliasAndRename() throws Exception {
+        __testRestoreWithRename(false, false, false, false, false);
+    }
+
+    public void testRestoreOverExistingOpenWithAliasAndRename() throws Exception {
+        __testRestoreWithRename(true, false, true, true, true);
+    }
+
+    public void testRestoreOverExistingOpenWithoutAliasAndWithRename() throws Exception {
+        __testRestoreWithRename(true, false, false, true, true);
+    }
+
+    public void testRestoreOverExistingClosedWithAliasAndRename() throws Exception {
+        __testRestoreWithRename(true, true, true, true, true);
+    }
+
+    public void testRestoreOverExistingClosedWithoutAliasAndWithRename() throws Exception {
+        __testRestoreWithRename(true, true, false, true, true);
+    }
+
+    public void testRestoreOverExistingOpenWithoutAliasAndRename() throws Exception {
+        __testRestoreWithRename(true, false, false, false, false);
+    }
+
+    public void testRestoreOverExistingOpenWithAliasAndWithoutRename() throws Exception {
+        __testRestoreWithRename(true, false, true, false, false);
+    }
+
+    public void testRestoreOverExistingClosedWithoutAliasAndRename() throws Exception {
+        __testRestoreWithRename(true, true, false, false, false);
+    }
+
+    public void testRestoreOverExistingClosedWithAliasAndWithoutRename() throws Exception {
+        __testRestoreWithRename(true, true, true, false, false);
+    }
+
+    private void __testRestoreWithRename(boolean exists, boolean closed, boolean includeAlias, boolean renameAliases, boolean renameIndexes)
```

Comment:
I'm not fond of using underscores like this.  Are they really needed?

Reply:
I think I just did this so it wouldn't be treated as a test, but I don't think the underscores is required. I have removed them.

Reply:
I think since it has arguments it can't run as a test anyway.